components.json

```diff
@@ -85,7 +85,7 @@
 		"actionscript": {
 			"title": "ActionScript",
 			"require": "javascript",
-			"peerDependencies": "markup",
+			"modify": "markup",
```

Comment:
Maybe this is too much but what if we changed `modify` -> `modifies` & `require` -> `dependencies`? That has some nice symmetry internally as well as with other tooks (`dependencies` in pkg.json). Thoughts?

Reply:
Well, I wouldn't combine `modify` + `require`. Let's step back for a second: we have these three keys currently:

1. `modify` - languages that, if loaded before this language, will be modified by it.
2. `require` - languages that need to be loaded before this language.
3. `after` - languages that if loaded, should appear before this language (? - not sure about this)

Are these descriptions correct?

Reply:
Yes, they are correct. Both `modify` and `after` describe optional dependencies. Both `after` and `require` describe dependencies where we assume that the depended resource is only read and never changed.

| | normal dependency | optional dependency |
| --- | --- | --- |
| **readonly** | `require` | `after` |
| **read-write** | --- | `modify` |

Reply:
What about this minor adjustment then:

| | normal dependency | optional dependency |
| --- | --- | --- |
| **readonly** | `require` | `optional` |
| **read-write** | --- | `modify` |

Reply:
Totally for it. The only reason why I choose `after` is because some plugins call it that right now.

Reply:
Were those plugins updated as well?

Reply:
[Yes.](https://github.com/PrismJS/prism/blob/39ff76e3ba7300ef762551a9c8fd8e030a272323/components.json#L1185)